exp/api/v1beta1/ocimachinepool_types.go

```diff
@@ -0,0 +1,172 @@
+/*
+ *
+ * Copyright (c) 2022, Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * /
+ *
+ */
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/errors"
+)
+
+// +kubebuilder:object:generate=true
+// +groupName=infrastructure.cluster.x-k8s.io
+
+// Constants block.
+const (
+	// MachinePoolFinalizer is the finalizer for the machine pool.
+	MachinePoolFinalizer = "ocimachinepool.infrastructure.cluster.x-k8s.io"
+)
+
+// OCIMachinePoolSpec defines the desired state of OCIMachinePool
+type OCIMachinePoolSpec struct {
+	// ProviderID is the ARN of the associated InstancePool
+	// +optional
+	ProviderID *string `json:"providerID,omitempty"`
+
+	// OCID of the image to be used to launch the instance.
+	ImageId string `json:"imageId,omitempty"`
+
+	// MinSize defines the minimum size of the group.
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=1
+	MinSize int32 `json:"minSize"`
+
+	// MaxSize defines the maximum size of the group.
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=1
+	MaxSize int32 `json:"maxSize"`
+
+	// Custom metadata key/value pairs that you provide, such as the SSH public key
+	// required to connect to the instance.
+	Metadata map[string]string `json:"metadata,omitempty"`
+
+	// The shape configuration of rhe instance, applicable for flex instances.
+	ShapeConfig ShapeConfig `json:"shapeConfig,omitempty"`
+
+	//TODO Add
```

Comment:
is this todo still valid?

Reply:
So the support for multiple ADs is still valid, I'm not sure if we see allowing a machine pool to be on a different subnet. I'll remove the others.